mavros_extras/src/plugins/mount_control.cpp

```diff
@@ -43,16 +44,34 @@ class MountControlPlugin : public plugin::PluginBase {
 		PluginBase::initialize(uas_);
 
 		command_sub = mount_nh.subscribe("command", 10, &MountControlPlugin::command_cb, this);
+		mount_orientation_pub = mount_nh.advertise<geometry_msgs::Quaternion>("orientation", 10);
 	}
 
 	Subscriptions get_subscriptions()
 	{
-		return { /* Rx disabled */ };
+		return {
+			make_handler(&MountControlPlugin::handle_mount_orientation)
+		};
 	}
 
 private:
 	ros::NodeHandle mount_nh;
 	ros::Subscriber command_sub;
+	ros::Publisher mount_orientation_pub;
+
+	/**
+	 * @brief Publish the mount orientation
+	 *
+	 * Message specification: https://mavlink.io/en/messages/common.html#MOUNT_ORIENTATION
+	 * @param o	received MountOrientation msg
+	 */
+	void handle_mount_orientation(const mavlink::mavlink_message_t *msg, mavlink::common::msg::MOUNT_ORIENTATION &o)
+	{
+		auto q = ftf::quaternion_from_rpy(Eigen::Vector3d(o.roll, o.pitch, o.yaw) * M_PI / 180.0);
```

Comment:
I am a little confused on which frame this quaternion represents. The yaw is relative to the vehicle, but roll and pitch is in the global frame. Shouldn't it use yaw_absolute to be more consistent?

Reply:
Actually, I think it's ok as it is, because it returns the same rpy values that you'd send with MAV_CMD_DO_MOUNT_CONTROL. In our usecase we have a gimbal with a yaw range of -320 to 320 degrees and we would like to get that value, no matter how the drone is positioned.

Reply:
I understand that the relative yaw values are more useful than the absolute value, but the frame you are publishing as orientation is not properly defined anywhere. What are the two frames orientation `q` is actually representing?

Reply:
Thanks for the reminder! According to mavlink description, the roll and pitch and yaw_absolute are according to global frame. roll, pitch and yaw would not be in the same frame. Maybe we can publish this similar to MountControl.msg: as roll, pitch, yaw, yaw_absolute in a MountOrientation.msg and not as quaternion?

Reply:
@dayjaby I am a little hesitant on having a new custom message definition for that, but I do think that would be less confusing.

Reply:
Then let's publish roll, pitch and yaw_absolute as a quaternion. The relative yaw can be reconstructed if needed by subtracting the heading (/mavros/global_position/compass_hdg). This might be troublesome though, because PX4 only sends MOUNT_ORIENTATION.yaw but no MOUNT_ORIENTATION.yaw_absolute (see https://github.com/PX4/Firmware/blob/c75954fef8ec6a7c7e4f7a68b2e536c1b5b633ab/src/modules/mavlink/mavlink_messages.cpp#L4722). I will do a PR for that to set yaw_absolute as well. This doesn't matter for gimbals who construct complete MOUNT_ORIENTATION messages, but at least for the typhoon h480 in sitl.

Reply:
Ooh I didn't see that coming. Then I guess doing it as the initial implementation makes sense. I think we just need to be careful not to forget what the yaw means in that case.